data.py

```diff
@@ -0,0 +1,83 @@
+import csv
+import json
+import operator
+
+import nlu
+import questions
+
+
+class Similarity(object):
+    def __init__(self, intent: str, score: float):
+        self.intent = intent
+        self.score = score
+
+    def __repr__(self):
+        return '<Similarity {intent} {score}>'.format(
+            intent=self.intent, score=self.score)
+
+
+class SimilarityResult(object):
+
+    def __init__(self, query_text: str, intents: [Similarity]):
+        self.query = query_text
+        self.intents = intents
+
+    def __repr__(self):
+        return '<SimilarityResult query {query} topScoringIntent {topScoringIntent} intents {intents}>'.format(
+            query=self.query, topScoringIntent=self.topScoringIntent, intents=self.intents)
+
+    @property
+    def intents(self):
+        return self._intents
+
+    @intents.setter
+    def intents(self, intents: [Similarity]):
+        self._intents = intents
+        self.topScoringIntent = max(
+            self.intents, key=operator.attrgetter('score'))
+
+    def to_json(self):
+        return json.dumps(
+            self,
+            default=lambda o: o.__dict__,
+            ensure_ascii=False)
+
+
+def from_csv_file(path: str) -> [questions.ExampleQuestion]:
+    with open(path, 'r', encoding='utf-8') as f:
```

Comment:
I'm thinking why we do not return `open(path, 'r', encoding='utf-8')`? Any ideas? Seems `with` is no need.

Reply:
without `with` means we need to close the file manually at the end of function.

Reply:
https://stackoverflow.com/questions/1832528/is-close-necessary-when-using-iterator-on-a-python-file-object

According to the answer, GC would collect it, but `with` is a better practice. So we keep it.